File: bash/js/models/Runner.js
const components = require('../uniflow/components')
const execSh = require('exec-sh')

function Runner() {

}

Runner.prototype.run = function(stack) {
    let commands = [], runner = {
        eval: function (code) {
            if (code === undefined) return

            commands.push(code)
        }
    }

    return stack.reduce((promise, item) => {
        return promise.then(() => {
            let component = new components[item.component]()
            component.deserialise(item.data)
            return component.onExecute(runner)
        })
    }, Promise.resolve())
        .then(() => {
            return new Promise(function(resolve, reject) {
                execSh(commands.join("\n"), {}, function(err, stdout, stderr) {
                    if(err) {
                        reject(stderr)
                    } else {
                        resolve(stdout)
                    }
                })
            })
        }).then(function(stdout) {
            process.stdout.write(stdout)
        })
}

module.exports = Runner

File: idea-plugin/src/main/java/fr/darkwood/uniflow/models/Runner.java
package fr.darkwood.uniflow.models;

import com.google.gson.JsonArray;
import com.google.gson.JsonObject;
import fr.darkwood.uniflow.components.Javascript;

import java.util.Iterator;

public class Runner {
    public void run(JsonArray stack) {
        for (int i = 0; i < stack.size(); i++) {
            JsonObject item = stack.get(i).getAsJsonObject();

            String componentClass = item.get("component").getAsString();

            if(componentClass.equals("javascript")) {
                String data = item.get("data").getAsString();
                Javascript component = new Javascript();
                component.deserialise(data);
                component.onExecute();
            }
        }
    }
}
